.circleci/config.yml

```diff
@@ -0,0 +1,265 @@
+    #
+    # TEST_SITE_NAME: The name of the test site to provide when installing.
+    # ADMIN_PASSWORD: The admin password to use when installing.
+    # ADMIN_EMAIL:    The email address to give the admin when installing.
+    #=========================================================================
+    TZ: "/usr/share/zoneinfo/America/Los_Angeles"
+
+    # The variables below usually do not need to be modified.
+    
+    #======================================================================================================================================
+    # Circle CI 2.0 does not yet expand environment variables so they have to be manually EXPORTed
+    # Once environment variables can be expanded the variables below can be uncommented and the EXPORTs in set-up-globals.sh can be removed
+    # See: https://discuss.circleci.com/t/unclear-how-to-work-with-user-variables-circleci-provided-env-variables/12810/11
+    # See: https://discuss.circleci.com/t/environment-variable-expansion-in-working-directory/11322
+    # See: https://discuss.circleci.com/t/circle-2-0-global-environment-variables/8681
+    #======================================================================================================================================
+    #BRANCH: $(echo $CIRCLE_BRANCH | grep -v '^\(master\|[0-9]\+.x\)$')
+    #PR_ENV: ${BRANCH:+pr-$BRANCH}
+    #CIRCLE_ENV: ci-$CIRCLE_BUILD_NUM
+    #DEFAULT_ENV: $(echo ${PR_ENV:-$CIRCLE_ENV} | tr '[:upper:]' '[:lower:]' | sed 's/[^0-9a-z-]//g' | cut -c -11 | sed 's/-$//')
+    #TERMINUS_ENV: ${TERMINUS_ENV:-$DEFAULT_ENV}
+    #PATH: ${PATH}:${HOME}/terminus/bin:${HOME}/bin
+    
+    NOTIFY: 'scripts/github/add-commit-comment {project} {sha} "Created multidev environment [{site}#{env}]({dashboard-url})." {site-url}'
+    ADMIN_USERNAME: admin
+    BUILD_TOOLS_VERSION: dev-master
+    TERM: dumb
+
+version: 2
+jobs:
+    # @todo: common initialization, maybe
+    build:
+        <<: *defaults
+        steps:
+            - checkout
+            - restore_cache:
+                keys:
+                    - composer-cache
+                    - terminus-install
+
+            - run:
+                name: dependencies
+                command: ./.circleci/set-up-globals.sh
+
+            - run:
+                name: build assets
+                command: ./.circleci/build-assets.sh
+
+            - save_cache:
+                key: composer-cache
+                paths:
+                    - $HOME/.composer/cache
+            
+            - save_cache:
+                key: terminus-install
+                paths:
+                    - $HOME/terminus
+                    - $HOME/.terminus
+
+            - persist_to_workspace:
+                # Must be relative path from working_directory
+                root: .
+                # Must be relative path from root
+                paths:
+                    - vendor
+                    - web
+
+    code_sniff:
```

Comment:
Code sniffing should run in one or two seconds. Do we get much benefit from having it be a separate job from unit tests?